### PR TITLE
fix: bulk update invoice remarks during site upgrade

### PR DESCRIPTION
**Pull request by vv-varun**
Upgrade to v15.35 onwards fails in site migration step : Executing erpnext.patches.v15_0.update_invoice_remarks

Fixes issue https://github.com/frappe/erpnext/issues/43634

### CODE REVIEW

**Comment by ruthra-kumar** [comment]
@vorasmit Can you take a look?

File: erpnext/patches/v15_0/update_invoice_remarks.py [hunk 1]
```diff
@@ -8,44 +8,144 @@ def execute():
...
+
+
+def update_sales_invoice_gle_remarks():
+
+	##	Update query to update GL Entry - Updates all entries which are for Sales Invoice with No Remarks 
+	update_query = """
+		UPDATE 
+		`tabGL Entry` as gle
+		INNER JOIN `tabSales Invoice` as si 
+		ON gle.voucher_type = 'Sales Invoice' AND gle.voucher_no = si.name AND gle.remarks = %(remarks)s
```
Comment (ruthra-kumar) [inline erpnext/patches/v15_0/update_invoice_remarks.py:82]: Joining GL Entry with Sales Invoice will surely be an issue on large databases.

Would also not recommend using 'remarks' on join condition, as it is a 'Text' field.

Comment (vv-varun) [reply erpnext/patches/v15_0/update_invoice_remarks.py:82]: Any better ideas on how to update this data ? 

The existing script (the one that was shipped with v15.35) always fails.. we have a live example of this... 

Moreover, the above queries work - I have executed these queries on my system.. In my system it took ~5 sec to execute the query. 
There are about ~100K invoices in my system. 

On a side note - this data migration should be optional. This is merely updating comments / remarks.

Comment (vv-varun) [reply erpnext/patches/v15_0/update_invoice_remarks.py:82]: @ruthra-kumar @vorasmit - Could you please review and close this PR.

Comment (vv-varun) [reply erpnext/patches/v15_0/update_invoice_remarks.py:82]: @ruthra-kumar @vorasmit - any update on this PR ? 

Let me know if there are any alternate approaches to resolving the upgrade issue.

Comment (ruthra-kumar) [reply erpnext/patches/v15_0/update_invoice_remarks.py:82]: @vv-varun 

Will review and update it by this week

**Comment by ruthra-kumar** [comment]
Query planner shows `voucher_type_voucher_no_index` index will be used.

```
+------+-------------+-------+--------+------------------------------------------+-------------------------------+---------+----------------------------------+------+-------------+
| id   | select_type | table | type   | possible_keys                            | key                           | key_len | ref                              | rows | Extra       |
+------+-------------+-------+--------+------------------------------------------+-------------------------------+---------+----------------------------------+------+-------------+
|    1 | SIMPLE      | gle   | ref    | voucher_no,voucher_type_voucher_no_index | voucher_type_voucher_no_index | 563     | const                            | 8    | Using where |
|    1 | SIMPLE      | si    | eq_ref | PRIMARY                                  | PRIMARY                       | 562     | _051d81649484759f.gle.voucher_no | 1    | Using where |
+------+-------------+-------+--------+------------------------------------------+-------------------------------+---------+----------------------------------+------+-------------+
```

**Comment by ruthra-kumar** [comment]
@vv-varun 

What was the RAM size on the your server where you tested this?

**Comment by ruthra-kumar** [comment]
Rebased on latest v15 hotfix

**Comment by vv-varun** [comment]
> @vv-varun
> 
> What was the RAM size on the your server where you tested this?

@ruthra-kumar  - I tested it in TEST and PROD systems both.
Our PROD runs on dedicated Frappe Cloud Server.
Both App and DB servers are 8GB RAM systems.

**Comment by ruthra-kumar** [comment]
@mergify copy develop

**Comment by vorasmit** [comment]
This could be useful going forward.
https://github.com/frappe/frappe/pull/28483

**Pull request merged**